### ISSUE TITLE
[Bug] New audio files sync'ed to my phone do not appear after sync has finished

### ISSUE THREAD

**Issue by ralhei** [SimpleMobileTools/Simple-Music-Player#404]
We are using SyncThing for to sync'ing images and music files btw. our mobile phones and a local NAS at home. Although those files are properly transferred to my mobile phone (and I can see them in the file browser on my phone), it often takes quite a while until they are visible in the corresponding folder in the music player. The music player itself reports the correct number of files in a folder (so it gets increased properly after a file was added by SyncThing), but the new audio file is not shown inside the folder, it can take up to 20mins sometimes (guessed number) until I see it.
Restarting the audio player sometimes helps, but not always, this seems to be random.

**Comment by tibbi** [comment]
do other music players work better?

**Comment by ralhei** [comment]
Hmm, hard to say, simple-music-player is the one and only player on my phone :-)  Would need to install a another player first before I can tell the difference ...

**Comment by tibbi** [comment]
what app and OS version are you using at all?

**Comment by ralhei** [comment]
I have two phones available, one is a Samsung with Android 11. This is were the update of the folder takes randomly long, and a restart of the player does not necessarily help.

The other phone runs GrapheneOS Version 12. Here the folders are also not updated while the player is running, but new files are immediately visible upon restart of the player. I guess a 'rescan music folders' button could maybe help, at least I would conclude this from the fact that at a restart those new music files are visible immediately.

**Comment by MaxsimLionel** [comment]
I have the same issue. New music downloaded with Spotiflyer shows up in Vania Music instantly, but takes 20 or so minutes for it to be playable and searchable in SMP.

**Comment by cboltz** [comment]
I have a similar issue, "just" with a completely different delay: I use scp with SimpleSSHD to copy podcasts to my phone - and it takes several days (!) until the new mp3 files appear in Simple Music Player. Even if I force-kill the app and delete cache _and_ app data. The outdated file list even survives a reboot of the phone. (In case it matters: Redmi 10C with Android 12)

Other music players I tested also have some delay, but at least a force-kill causes them to refresh the file list on the next start.

Long story short - please add an option to refresh the folder content.

**Comment by Naveen3Singh** [comment]
Are you guys still experiencing this issue?

**Comment by Naveen3Singh** [comment]
> Even if I force-kill the app and delete cache and app data. The outdated file list even survives a reboot of the phone.

@cboltz that's usually not possible. Simple Music Player's local cache is cleared as soon as you clear app data so it must show the newly indexed music files just like any other music player app. It could be that your particular issue is device-specific. Can you check this again with Simple Music Player and some other player?

**Comment by cboltz** [comment]
Tested again: I copied several mp3 files to my phone using scp with SimpleSSHD (into 3 different folders). Today (= a day later) they didn't show up in Simple Music Player, but this time also not in another music player. Cleaning the cache + force-killing both players didn't help. (Last time, force-killing the other player had helped to get it updated. Unfortunately I don't know what was different this time.)

This time, a reboot helped to get most (not all) folders updated. The other player lists all files. Simple Music Player has 2 of 3 folders updated, but still doesn't offer the new files in one of the folders. (The affected folder is the main "Music" folder, the two better-working folders are sub-folders of it.) There was also a java error message displayed, but unfortunately it went away before I could read or save it.

For comparison: The files showed up instantly in the file manager, and I could even play them by clicking them there and opening them in Simple Music Player.

This might indeed be a device-specific issue with folder refresh - and I still think offering a refresh option would be helpful ;-)

**Comment by cboltz** [comment]
Minor update to my previous comment: Several hours after the reboot, the main "Music" folder was also updated.

**Comment by Axeltherabbit** [comment]
Same issue, can we just add a button to manually refresh the songs? it always takes several minutes

**Comment by Naveen3Singh** [comment]
Hello! @cboltz @ralhei @MaxsimLionel @Axeltherabbit 

I have uploaded a test APK with potential fixes for each of your issues, please check if you are interested:
https://github.com/Naveen3Singh/Simple-Music-Player/releases/download/improved_media_scan/music-player-prepaid-debug.apk

Let me know if it works for any of you.

**Comment by tibbi** [comment]
should be fixed in 5.17.0
